pkg/perf/cpubench_task.go

```diff
@@ -0,0 +1,75 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"github.com/threefoldtech/zos/pkg/stubs"
+)
+
+// CPUBenchmarkTask defines CPU benchmark task data.
+type CPUBenchmarkTask struct {
+	taskID   string
+	schedule string
+}
+
+// CPUBenchmarkResult holds CPU benchmark results with the workloads number during the benchmark.
+type CPUBenchmarkResult struct {
+	Single    float64 `json:"single"`
+	Multi     float64 `json:"multi"`
+	Threads   int     `json:"threads"`
+	Workloads int     `json:"workloads"`
+}
+
+var _ Task = (*CPUBenchmarkTask)(nil)
+
+// NewCPUBenchmarkTask returns a new CPU benchmark task.
+func NewCPUBenchmarkTask() CPUBenchmarkTask {
+	return CPUBenchmarkTask{
+		taskID:   "CPUBenchmark",
+		schedule: "0 0 */6 * * *",
+	}
+}
+
+// ID returns task ID.
+func (c *CPUBenchmarkTask) ID() string {
+	return c.taskID
+}
+
+// Cron returns task cron schedule.
+func (c *CPUBenchmarkTask) Cron() string {
+	return c.schedule
+}
+
+// Run executes the CPU benchmark.
+func (c *CPUBenchmarkTask) Run(ctx context.Context) (interface{}, error) {
+	cpubenchOut, err := exec.CommandContext(ctx, "cpubench", "-j").CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute cpubench command: %w", err)
+	}
+	cpubenchData := struct {
```

Comment:
Why do we need another anonymous struct while we have CPUBenchmarkResult?

Reply:
as the JSON output does not have the `workloads` field and I don't know what the behavior of `json` package with missing fields.

Reply:
Hmm I think you can just do `json:"-"` for the field 

https://go.dev/play/p/K6b-FOSs6e3

Reply:
I can't use that because it will be ignored when querying the task data. I can just remove the anonymous struct and `json` ,I believe, will set it to the default value: 0